[Simplified Login] Prologue screen changes

hafizrahman opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #7610 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR implements the following Prologue screen designs (these are based on the iOS version):

| simplified login variant | control variant |
|-|-|
| <img src="https://user-images.githubusercontent.com/266376/199205250-d103a43d-7345-42bd-b561-92787b864a97.png" /> | <img src="https://user-images.githubusercontent.com/266376/199204627-3a6e0cd8-f9c1-445b-9b85-8d3bdc22976b.png" /> |

The PR includes various small design updates, such as:
1. Making the design work on smaller screen (tested with a 4" screen)
2. updates to the sw600dp and landscape layouts
3. the "OR" divider on control layout, above the "Get started" button

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

Run app in logged out state and compare the shown prologue screen with the design above.

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hafizrahman committed:
Hide "Login with Site Address" button if in Simplified Login variant.

hafizrahman committed:
Update "New to Woo" button in Simplified Login variant.

hafizrahman committed:
Add string for "Create a Store" button.

hafizrahman committed:
Add: Create A Store button in default layout.

hafizrahman committed:
Add: Create A Store button in sw600dp layout.

hafizrahman committed:
Add: Create A Store button in landscape layout.

hafizrahman committed:
Show Create A Store button in Simplified Login variant.

This also hides the "Login With Site Address" and "Get Started" buttons.

hafizrahman committed:
Only show Create A Store button if its feature flag is enabled.

hafizrahman commented on WooCommerce/src/main/res/layout/fragment_login_prologue.xml:
```diff
@@ -69,45 +69,61 @@
-    <com.google.android.material.button.MaterialButton
-        android:id="@+id/button_login_wpcom"
-        style="@style/Woo.Button.Colored"
+    <LinearLayout
```
The initial layout was a bit complex to my brain with the differing buttons's placements / constraint positioning. 

With how some buttons are now shown and hidden depending on the variant, I thought it'll be easier to follow when everything is put in a LinearLayout.

hichamboushaba replied:
Yes, when possible, a LinearLayout is always better and more performant than a ConstraintLayout.

The initial layout was actually broken 😄, it caused some issues like this p1666951087613899-slack-C70PAM3RA, so thanks for updating it.

hichamboushaba replied:
When I replied above, I assumed you changed the root layout to LinearLayout too.
I think since you moved the buttons to their own layout, it's better to change the root layout too, it'll make things clearer and easier to follow, and it'll make handling the resizing of the image easier when using weights. just a suggestion 🙂.

hafizrahman commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt:
```diff
@@ -60,6 +66,16 @@ open class LoginPrologueFragment(@LayoutRes layout: Int) : Fragment(layout) {
+            binding.buttonLoginStore.hide()
+            binding.buttonGetStarted.hide()
+
+            binding.buttonCreateStore.isVisible = FeatureFlag.STORE_CREATION_WEBVIEW_FLOW.isEnabled()
```
Button doesn't do anything for now, let's make separate PR to make it open to the right flow as I haven't checked the status / availability for it.

hichamboushaba replied:
As I explained below, the "Create Store" and "Get Started" should be a single button, and it depends on `FeatureFlag.ACCOUNT_CREATION_FLOW`.

hafizrahman committed:
Add Simplified Login variant's prologue screen subtitle.

hafizrahman committed:
Replace login button text to "Log In" for Simplified Login variant.

hafizrahman committed:
Update release notes.

hichamboushaba commented on WooCommerce/src/main/res/layout/fragment_login_prologue.xml:
```diff
@@ -59,55 +59,90 @@
-        android:textAllCaps="false"
-        android:visibility="gone"
-        app:layout_constraintBottom_toBottomOf="parent" />
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/button_create_store"
+            style="@style/Woo.Button.Colored.White"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/major_100"
+            android:layout_marginEnd="@dimen/major_100"
+            android:layout_marginBottom="@dimen/major_150"
+            android:text="@string/login_create_store"
+            android:textAllCaps="false"
+            android:visibility="gone" />
+
+        <com.google.android.material.button.MaterialButton
+            android:id="@+id/button_get_started"
+            style="@style/Woo.Button.Colored.White"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_marginStart="@dimen/major_100"
+            android:layout_marginEnd="@dimen/major_100"
+            android:layout_marginBottom="@dimen/major_150"
+            android:text="@string/signup_get_started_button"
+            android:textAllCaps="false"
+            android:visibility="gone" />
```
I think we should have just a single button here, whether it's called "Get Started" or "Create a store" is TBD (check the comment here ERGtCFLyiC9CFpHsw1XIid-fi-411%3A6226), but it's the same button with the same behavior.

hafizrahman committed:
Merge branch 'trunk' into activate/7610-prologue-changes

hafizrahman committed:
Fix: Variant checking.

hafizrahman committed:
Add: force testing to use control variant.

e2e tests might break if using test variant, since some buttons/elements might be missing.

hafizrahman committed:
Fix: detekt spacing issue.

hafizrahman committed:
Update design to only have one button for Get Started/Create a Store.

hafizrahman committed:
Move Simplified Login test variant setup to its own function.

hafizrahman commented:
This is ready for another review @hichamboushaba :)

hafizrahman committed:
Activate starting event for Simplified Login A/B test.

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/config/FirebaseRemoteConfigRepository.kt:
```diff
@@ -94,7 +94,10 @@ class FirebaseRemoteConfigRepository @Inject constructor(
-        return try {
+        return if (PackageUtils.isTesting()) {
+            // Use control variant to prevent e2e test breakages
+            LoginVariant.valueOf(defaultValues[SIMPLIFIED_LOGIN_VARIANT_KEY]!!)
```
You are returning the simplified login variant here, I assume it's just an error, you should return the Control variant.

hafizrahman replied:
It's returning `defaultValues`, though, which should return Control 🤔

hichamboushaba replied:
Ah, sorry, you are right, I didn't read the code correctly here.

But still, just a np, I would use the `LoginVariant` directly here, instead of passing by the `defaultValue`, just in case we wanted to change the defaultValue in the future, it would be best to keep the UI tests value hardcoded to the correct value.

hafizrahman replied:
Good point!

hichamboushaba commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt:
```diff
@@ -32,9 +36,13 @@ open class LoginPrologueFragment(@LayoutRes layout: Int) : Fragment(layout) {
     private var prologueFinishedListener: PrologueFinishedListener? = null
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        simplifiedLoginExperiment.activate()
```
I think it's better to check the `savedInstanceState` value before triggering `activate`, just to make sure it's triggered only once.

hafizrahman replied:
We discussed this and decided it's OK for it to be run multiple times.

hafizrahman committed:
Ensure Simplified Login experiment is only activated once.

hafizrahman committed:
Hardcode CONTROL variant value instead in case default value changes in the future.

hafizrahman committed:
Merge branch 'trunk' into activate/7610-prologue-changes

hafizrahman committed:
Various design updates and improvements:

1. Illustration now scales depending on image screen and does not take too much time.

2. In turns, that makes all buttons and text readable even on smaller screen.

2. Title text and design are updated. It is now used both on Control and Simplified Login variant.

3. "New To WooCommerce" button is hidden on all variants.

4. "Get started" button is shown on both Control and Simplified Variant.

hafizrahman committed:
Updates to the landscape layout.

- Support for showing Get Started button.
- Update the title view.

hafizrahman committed:
Fix spacing.

hafizrahman committed:
Add: "OR" divider in the layout for control variant.

hafizrahman committed:
Fix Prologue title positioning to stick closer to the illustration even in different screen heights.

hafizrahman committed:
Add: "OR" divider on sw600dp layout.

hafizrahman committed:
Add: "OR" divider on landscape layout.

hafizrahman commented on WooCommerce/src/main/res/layout-land/fragment_login_prologue.xml:
```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Since we don't have actual design for this, I'm basically just updating this based on what I feel would be a good enough design.

hafizrahman commented on WooCommerce/src/main/res/layout-sw600dp/fragment_login_prologue.xml:
```diff
@@ -35,32 +35,35 @@
 
     <ImageView
         android:id="@+id/imageView"
```
Same as landscape layout: Since we don't have actual design for this, I'm basically just updating this based on what I feel would be a good enough design.

hafizrahman commented on WooCommerce/src/main/res/layout/fragment_login_prologue.xml:
```diff
@@ -9,105 +9,150 @@
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@id/prologueTitle" />
+        app:layout_constraintTop_toBottomOf="@id/prologueTitle"
+        android:visibility="gone" />
```
`newToWooButton` is now supposed to be hidden both on control and prologue, but I'm choosing not to delete it outright yet just in case it needs to be re-added later.

hichamboushaba replied:
Personally, I think we should remove it, a good read on why https://kentcdodds.com/blog/please-dont-commit-commented-out-code

hafizrahman replied:
Good point. I removed it.

hafizrahman committed:
Update binding call now that divider exists in all types of layout.

hafizrahman committed:
Fix constraint.

hafizrahman committed:
Remove all "New to Woo" button(s) and code related to it.

hafizrahman committed:
Merge branch 'trunk' into activate/7610-prologue-changes

hichamboushaba commented on WooCommerce/src/main/res/values/strings.xml:
```diff
@@ -204,6 +204,7 @@
     <string name="login_with_store_address">Log in with your store address</string>
     <string name="login_configure_link">Read the %1$sconfiguration instructions%2$s.</string>
     <string name="login_with_qr_code">Scan QR code to login</string>
+    <string name="login_create_store">Create a Store</string>
```
I don't think this is used anywhere?

hafizrahman replied:
Thanks!

hichamboushaba reviewed (approved):
Nice job @hafizrahman 💯, I left one comment about one unused string resource, otherwise it's good to go.

hafizrahman committed:
Remove unused string.

hafizrahman committed:
Remove unused import.

hafizrahman committed:
Merge branch 'trunk' into activate/7610-prologue-changes
